iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -32,7 +32,15 @@ namespace roudi
 capro::Interfaces StringToCaProInterface(const capro::IdString_t& str) noexcept
 {
     int32_t i{0};
-    convert::fromString(str.c_str(), i);
+    auto result = convert::from_string<int32_t>(str.c_str());
+    if (!result.has_value())
+    {
+        IOX_LOG(WARN, "conversion failure");
+        return capro::Interfaces::INTERNAL;
+    }
+
+    i = result.has_value();
```

Comment:
Oh boy, we really need to fix the serialization/deserialization :sweat_smile: 

Furthermore, this is one of the oldest classes in iceoryx and dates back to the research department. This scream so much to just returning an `optional` or `expected`. Well, something for the future an not now.

Reply:
See [809dddc](https://github.com/eclipse-iceoryx/iceoryx/pull/2135/commits/809dddca229ad98fa46ad08ccfd271927ae83cab).